mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -252,7 +250,7 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
           model,
           Array(0D),
           Array(0D))
-        return copyValues(model.setSummary(trainingSummary))
+        return model.setSummary(trainingSummary)
```

Comment:
This is a minor bug of ```LinearRegression```, we should first copy values from parent estimator and then call ```findSummaryModelAndPredictionCol```, otherwise we will always get empty ```predictionCol```(and other params) for the ```LinearRegressionModel```.